docs/_newsfragments/1589.breakingchange.rst

```diff
@@ -0,0 +1,17 @@
+General refactoring of internal media handler:
+
+*  Deserializing an empty body with an handler that does not support it will
+   raise :class:`falcon.MediaNotFoundError`, and return a
+   ``400 Bad Request`` response. This error may be suppressed by passing
+   a default value to ``get_media`` to be used in case of empty body.
+   See also :meth:`Request.get_media` for details.
+   Previously ``None`` was returned in all cases without calling the handler.
+*  Exceptions raised by the handlers are wrapped as
+   :class:`falcon.MediaMalformedError`, and return a
+   ``400 Bad Request`` response.
+*  Subsequent calls to :meth:`Request.get_media` or :attr:`Request.media` will
+   re-raise the same exception, if the fist call ended in an error, unless the
+   exception was a :class:`falcon.MediaNotFoundError` and a default value is passed.
```

Comment:
I am not 100%, but I would add a comma here: exception was a <class>, and a default value is passed.

Reply:
I think not, since I want to connect the exception with the default being passed. Alone it makes sense without: "unless the exception was a :class:`falcon.MediaNotFoundError` and a default value is passed."

Reply:
Either way is absolutely fine for me. _methinks_ though it is a compound sentence with two clauses: the **exception was** a class, and a default **value is** passed.

Reply:
I view the compound as the before part: _if the fist call ended in an error_

Reply:
It could very well be that you are right. I normally just let @kgriffs fix stuff like this for me. :smiling_imp: